message HMMER_Request {
  // Query sequence in FASTA format
  required string fasta_sequence = 1;
}

message HMMER_Response {
  message Alignment {
    required string template_id = 1          [ default = "blank" ];
    required uint32 query_start = 2          [ default = 0 ];
    required uint32 template_start = 3       [ default = 0 ];
    required string aligned_query_seq = 4    [ default = "" ];
    required string aligned_template_seq = 5 [ default = "" ];
    optional float log_evalue = 6            [ default = 0 ];
    optional float bit_score = 7             [ default = 0 ];
  }

  // Unique identifier for the request
  required uint64 job_id = 1;
  repeated Alignment alignments = 2;
}
